File: Codeforces/div805/B.cpp

#include <iostream>
#include <set>
#include <string>

using namespace::std;

int main()
{
	int t;
	cin >> t;
	while (t--)
	{
		string a;
		cin >> a;
		set<int> s;
		int ans = 0;
		for (char c : a)
		{
			if (s.count(c) > 0)
				continue ;
			else if (s.size() == 3)
			{
				++ans;
				s.clear();
				s.insert(c);
			}
			else
				s.insert(c);
		}
		cout << ++ans << endl;
	}
}

File: Codeforces/div805/B.py

for _ in range(int(input())):
    a = input()
    s = set()
    ans = 0
    for i in a:
        if i in s:
            continue
        else:
            if len(s) == 3:
                ans += 1
                s = {i}
            else:
                s.add(i)
    print(ans + 1)

